packages/tailwindcss/src/index.ts

```diff
@@ -110,6 +112,18 @@ export async function compile(
         )
       }
 
+      // Figure out if the custom utility uses `@apply` so that we can
+      // substitute it later.
+      walk(node.nodes, (child) => {
+        if (child.kind !== 'rule') return
+        if (child.selector[0] !== '@') return
+        if (!child.selector.startsWith('@apply ')) return
+
+        customUtilitiesWithAtApply.push(node)
+
+        return WalkAction.Stop
```

Comment:
Once we know `@apply` is being used, we can stop early. We will handle all the `@apply` nodes later.